src/data/value.ts

```diff
@@ -1,6 +1,88 @@
 import { DateTime, Duration } from "luxon";
 import { DEFAULT_QUERY_SETTINGS, QuerySettings } from "settings";
 import { getFileName } from "util/normalize";
+// import { Field, BinaryOpField } from "expression/field";
+
+/** A specific task. */
+export class Task {
+    /** The text of this task. */
+    text: string;
+    /** The line this task shows up on. */
+    line: number;
+    /** The full path of the file. */
+    path: string;
+    /** Whether or not this task was completed. */
+    completed: boolean;
+    /** Whether or not this task and all of it's subtasks are completed. */
+    fullyCompleted: boolean;
+    /** If true, this is a real task; otherwise, it is a list element above/below a task. */
+    real: boolean;
+    /** Any subtasks of this task. */
+    subtasks: Task[];
+    /** An explicitly defined date of completion */
+    completedDate?: DateTime;
+    /** An explicitly defined date of creation */
+    createdDate?: DateTime;
+    /** An explicitly defined due date */
+    dueDate?: DateTime;
+    /** The block this task shows up on */
+    blockId: string;
+
+    constructor(init?: Partial<Task>) {
+        Object.assign(this, init);
+        this.subtasks = (this.subtasks || []).map(t => new Task(t));
+        this.completedDate = init?.completedDate;
+        this.createdDate = init?.createdDate;
+        this.dueDate = init?.dueDate;
+    }
+
+    public link(): string {
+        return `${this.path}#${this.blockId}`;
+    }
+
+    public id(): string {
+        return `${this.path}-${this.line}`;
+    }
+
+    public markdown(): string {
+        let bulletChar = `-`;
+        let stateChar = this.completed ? "x" : " ";
+        let result = `${bulletChar} [${stateChar}] ${this.text}`;
+        return result;
+    }
```

Comment:
One thing to keep in mind - this will create a checkbox but it will not be checkable/uncheckable. That's fine for now, though in the long run it would be very interesting to make it so it works the same as a task list does, but anywhere.

Reply:
No action required on this right now, though.